src/app.rs

```diff
@@ -1264,6 +1272,20 @@ impl Application for App {
                 config_set!(app_theme, app_theme);
                 return self.update_config();
             }
+            Message::Compress(entity_opt) => {
+                let paths = self.selected_paths(entity_opt);
+                if let Some(current_path) = paths.get(0) {
+                    if let Some(destination) = current_path.parent().zip(current_path.file_stem()) {
+                        let to = destination.0.to_path_buf();
+                        self.dialog_pages.push_back(DialogPage::Compress {
+                            paths,
+                            to,
+                            name: String::new(),
```

Comment:
It would be nice to pre-fill the name. If there is only one path selected, it should probably be that path with its extension replaced with .zip